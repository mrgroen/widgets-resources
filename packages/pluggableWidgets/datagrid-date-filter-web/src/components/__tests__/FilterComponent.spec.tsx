import { shallow } from "enzyme";
import { createElement } from "react";
import { FilterComponent } from "../FilterComponent";

describe("Filter component", () => {
    it("renders correctly", () => {
        const component = shallow(<FilterComponent adjustable defaultFilter="equal" filterDispatcher={jest.fn()} />);

        expect(component).toMatchSnapshot();
    });

    it("renders correctly when not adjustable by user", () => {
        const component = shallow(
            <FilterComponent adjustable={false} defaultFilter="equal" filterDispatcher={jest.fn()} />
        );

        expect(component).toMatchSnapshot();
    });

    it("renders correctly with aria labels", () => {
        const component = shallow(
            <FilterComponent
                adjustable
                screenReaderButtonCaption="my label"
                screenReaderInputCaption="my label"
                defaultFilter="equal"
                filterDispatcher={jest.fn()}
            />
        );

        expect(component).toMatchSnapshot();
    });
});
