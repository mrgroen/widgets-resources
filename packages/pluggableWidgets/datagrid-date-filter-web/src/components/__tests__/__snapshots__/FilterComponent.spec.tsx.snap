// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component renders correctly 1`] = `
<div
  className="filter-container"
  data-focusindex={0}
>
  <FilterSelector
    defaultFilter="equal"
    onChange={[Function]}
  />
  <ForwardRef
    adjustable={true}
    setValue={[Function]}
    value={null}
  />
</div>
`;

exports[`Filter component renders correctly when not adjustable by user 1`] = `
<div
  className="filter-container"
  data-focusindex={0}
>
  <ForwardRef
    adjustable={false}
    setValue={[Function]}
    value={null}
  />
</div>
`;

exports[`Filter component renders correctly with aria labels 1`] = `
<div
  className="filter-container"
  data-focusindex={0}
>
  <FilterSelector
    ariaLabel="my label"
    defaultFilter="equal"
    onChange={[Function]}
  />
  <ForwardRef
    adjustable={true}
    screenReaderInputCaption="my label"
    setValue={[Function]}
    value={null}
  />
</div>
`;
